### PR TITLE
feat: Brush up package quality (Critical + High)

### PR DESCRIPTION
**Pull request by genzou9201**
## Summary
- **Dead code removal**: Delete unused `CalendarStore` class and `CalendarOutlineStyle` class
- **Magic number extraction**: Replace all hardcoded numeric values with named constants across view files
- **Test coverage expansion**: Add 10 new test cases (date selection logic, edge cases, double month layout) — 8 → 17 tests
- **Theme support**: Replace hardcoded `Colors.white/black/black54/grey` with `Theme.of(context).colorScheme` for dark mode compatibility
- **Accessibility**: Add `Semantics` to day cells and navigation buttons, `ExcludeSemantics` on weekday labels
- **README enhancement**: Add pub.dev badges, installation guide, feature list, and license section
- **pubspec.yaml cleanup**: Add `topics`, `repository`, `issue_tracker`; remove unused `mockito`/`build_runner` dev dependencies

## Test plan
- [x] `flutter analyze` — no issues
- [x] `flutter test` — all 17 tests passed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,77 +1,83 @@
...
+  return Scaffold(
+    body: Center(
+      child: DateRangeCalendar(
+        onTappedDay: onTappedDay,
       ),
-    );
-  }
+    ),
+  );
+}
```
[the remark this answers is not in the document]
Comment (genzou9201) [reply README.md:49]: 対応済み
Basic/Custom 両方を StatelessWidget でラップし、import も追加しました

File: README.md [hunk 2]
```diff
@@ -1,77 +1,83 @@
...
+        onTappedDay: onTappedDay,
+        setupData: mySetupData,
       ),
-    );
-  }
-```
\ No newline at end of file
+    ),
+  );
+}
```
[the remark this answers is not in the document]
Comment (genzou9201) [reply README.md]: 対応済み
Basic/Custom 両方を StatelessWidget でラップし、import も追加しました
